devops/providers/azure-devops/templates/infrastructure/azure-pipeline-build-stage.yml

```diff
@@ -61,8 +63,17 @@ jobs:
       inputs:
         version: '$(GO_VERSION)'
 
+    - task: Bash@3
+      name: SetupGitCredentialInjection
+      displayName: Configure Git to use PAT
+      condition: eq(${{ coalesce(config.overrideGitPAT, 'false') }}, 'true')
+      inputs:
+        targetType: 'inline'
+        script: '[[ -z $(TEMPLATE_REPO_PAT) ]] && echo "No PAT provided; skipping." || git config --global url."https://cobalt:$(TEMPLATE_REPO_PAT)@".insteadOf "https://"'
+
     - task: AzureCLI@1
       displayName: 'Unit Test Terraform Template'
+      condition: ne(${{ coalesce(config.skipTests, 'false') }}, 'true')
```

Comment:
Similar to above. Also check out the full list of functions if you haven't yet seen them:
https://docs.microsoft.com/en-us/azure/devops/pipelines/process/expressions?view=azure-devops#functions